### PR TITLE
Fix: parse later ES files in `eslint --init` (fixes #10003)

### LINKED ISSUE #10003: Parsing error: Unexpected token => when using eslint --init on post-ES2015 files
**Tell us about your environment**

* **ESLint Version:** 4.12.1
* **Node Version:** 9.5.0
* **npm Version:** 5.6.0

**What parser (default, Babel-ESLint, etc.) are you using?** Default

**Please show your full configuration:** N/A

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```js
(async () => {
});
```

```bash
eslint --init
```

Followed by answering prompts as follows:
- **File?** fileName.js
- **Format?** YAML
- **ES6 features?** Yes
- **ES6 modules?** Yes
- **Environment?** Node
- **JSX?** No

**What did you expect to happen?**

`.eslint.yaml` generated

**What actually happened? Please include the actual, raw output from ESLint.**

```bash
(~/tmp/example.js:1:11) Parsing error: Unexpected token =>
Error: (~tmp/example.js:1:11) Parsing error: Unexpected token =>
    at getSourceCodeOfFile (/usr/local/lib/node_modules/eslint/lib/util/source-code-util.js:38:15)
    at filenames.forEach.filename (/usr/local/lib/node_modules/eslint/lib/util/source-code-util.js:94:28)
    at Array.forEach (<anonymous>)
    at getSourceCodeOfFiles (/usr/local/lib/node_modules/eslint/lib/util/source-code-util.js:93:15)
    at configureRules (/usr/local/lib/node_modules/eslint/lib/config/config-initializer.js:165:23)
    at processAnswers (/usr/local/lib/node_modules/eslint/lib/config/config-initializer.js:288:18)
    at inquirer.prompt.then.secondAnswers (/usr/local/lib/node_modules/eslint/lib/config/config-initializer.js:536:32)
    at <anonymous>
Determining Config: 0% [------------------------------] 0.0s elapsed, eta 0.0s
```

### PR DESCRIPTION
**Pull request by g-plane**
**What is the purpose of this pull request? (put an "X" next to item)**

[x] Bug fix

close #10003 

**What changes did you make? (Give an overview)**
When running `eslint --init`, the CLI will ask user the ECMAScript version and whether using es6 globals.
Note that I haven't resolved ES7+ globals. This should be in another PR.

### CODE REVIEW

File: lib/config/config-initializer.js [hunk 1]
```diff
@@ -486,18 +489,35 @@ function promptUser() {
...
                 default: false,
                 when(answers) {
-                    return answers.es6 === true;
+                    return answers.ecmaVersion >= 2015;
+                }
+            },
+            {
+                type: "confirm",
+                name: "es6Globals",
+                message: "Are you using ES6 globals (such as `Map` and `Set`)?",
```
Comment (not-an-aardvark) [inline lib/config/config-initializer.js]: Nitpick: Can we call this "ES2015 globals", since it's referred to as "ES2015" in the previous question?

Comment (not-an-aardvark) [inline lib/config/config-initializer.js]: Can we reword this as "Do you want to enable..." rather than "Are you using..."? It seems like someone might answer "no" because they're currently not using `Map` and `Set`, even though they might want to do so in the future.

File: lib/config/config-initializer.js [hunk 2]
```diff
@@ -486,18 +489,35 @@ function promptUser() {
...
+                default: false,
+                when(answers) {
+                    return answers.ecmaVersion >= 2015;
```
Comment (not-an-aardvark) [inline lib/config/config-initializer.js]: What's the reasoning for only asking this question when `answers.ecmaVersion >= 2015`? If we're giving users the option to enable ES2015 parsing without ES2015 globals, it seems equally plausible that they might want to enable ES2015 globals without ES2015 parsing.

To me, it seems better to either guess the globals option based on the `ecmaVersion` answer (keeping things simpler for the user), or to always ask both questions. I would slightly prefer the former, since the `--init` helper is just supposed to address common cases and doesn't need to provide the same level of flexibility as creating a config file manually.

File: lib/config/config-initializer.js (hunk 1 above)
Comment (g-plane) [reply lib/config/config-initializer.js]: I'll will update it. BTW, this sentence is copied, not written by me.

File: lib/config/config-initializer.js (hunk 2 above)
Comment (g-plane) [reply lib/config/config-initializer.js]: So if the user choose "greater or equals to ES2015", skip this question and enable ES2015 globals?

Comment (not-an-aardvark) [reply lib/config/config-initializer.js]: Yes, that would be my recommendation.

Comment (g-plane) [reply lib/config/config-initializer.js]: Should I remove this question? I think, if the user choose ES3 or ES5, it shouldn't enable ES2015 globals, and if the user choose "greater or equals to ES2015", enable it by default.

Comment (not-an-aardvark) [reply lib/config/config-initializer.js]: Agreed, I think it would be best to remove the question and do what you just described.

**Comment by g-plane** [comment]
@not-an-aardvark Updated.

**Comment by platinumazure** [comment]
@not-an-aardvark Are your concerns addressed in this PR?
